### PR TITLE
Codacy fixes

### PR DESCRIPTION
**Pull request by AndyScherzinger**
needs https://github.com/nextcloud/android-library/pull/180

fixing several codacy warning categories, while for the for-loop I am respecting https://developer.android.com/training/articles/perf-tips :
> With an ArrayList, a hand-written counted loop is about 3x faster (with or without JIT), but for other collections the enhanced for loop syntax will be exactly equivalent to explicit iterator usage.

* codacy: String.indexOf(char) is faster than String.indexOf(String).
* codacy: Do not add empty strings
* codacy: Avoid the use of value in annotations when its the only element
* codacy: Avoid unnecessary constructors - the compiler will generate t…
* codacy: No need to check for null before an instanceof
* codacy: String.startsWith can be rewritten using String.charAt(0)
* codacy: A getX() method which returns a boolean should be named isX()
* codacy: No need to call String.valueOf to append to a string.
* codacy: Avoid using redundant field initializer
* codacy: This for loop can be replaced by a foreach loop

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky this PR contains the change needed for the merged lib change (isFavorite): https://github.com/nextcloud/android/pull/3070/commits/6b59ade7bde9233a840a57856e3747237b81c001 - This is a breaking change in the library

File: src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java [hunk 1]
```diff
@@ -2020,7 +2020,7 @@ private void onUpdateShareInformation(RemoteOperationResult result, @StringRes i
             snackbar.show();
         }
 
-        if (fileDetailFragment != null && fileDetailFragment instanceof FileDetailFragment) {
+        if (fileDetailFragment instanceof FileDetailFragment) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java]: getSecondFragment() could return null and thus null check is needed.
We should add @nullable as return

**Comment by tobiasKaminsky** [comment]
One null check should remain, otherwise :+1: 

Then let us merge it as soon as drone is green

File: src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java]: @tobiasKaminsky why would we need the nullCheck? `null instanceOf Whatever`should always be false, shouldn't it?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky see https://docs.oracle.com/javase/specs/jls/se10/html/jls-15.html#jls-15.20.2

> At run time, the result of the instanceof operator is true if the value of the RelationalExpression is not null and the reference could be cast to the ReferenceType without raising a ClassCastException. Otherwise the result is false.

File: src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java]: see https://github.com/nextcloud/android/pull/3070#issuecomment-424310161 for `null`it is `false`by language design :)

**Comment by tobiasKaminsky** [comment]
:+1:

**Pull request merged**